packages/flutter_tools/lib/src/migrations/lldb_init_migration.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:xml/xml.dart';
+import 'package:xml/xpath.dart';
+
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/project_migrator.dart';
+import '../build_info.dart';
+import '../ios/xcodeproj.dart';
+import '../project.dart';
+
+class LLDBInitMigration extends ProjectMigrator {
+  LLDBInitMigration(IosProject project, BuildInfo buildInfo, {required Logger logger})
+    : _xcodeProject = project,
+      _buildInfo = buildInfo,
+      _xcodeProjectInfoFile = project.xcodeProjectInfoFile,
+      super(logger);
+
+  final IosProject _xcodeProject;
+  final BuildInfo _buildInfo;
+  final File _xcodeProjectInfoFile;
+  String get initPath =>
+      _xcodeProject.lldbInitFile.path.replaceFirst(_xcodeProject.hostAppRoot.path, r'$(SRCROOT)');
+
+  @override
+  Future<void> migrate() async {
+    SchemeInfo? schemeInfo;
+    try {
+      if (!_xcodeProjectInfoFile.existsSync()) {
+        throw Exception('Xcode project not found.');
+      }
+
+      schemeInfo = await _getSchemeInfo();
+
+      final bool isSchemeMigrated = _isSchemeMigrated(schemeInfo);
+      if (isSchemeMigrated) {
+        return;
+      }
+      _migrateScheme(schemeInfo);
+    } on Exception catch (e) {
+      logger.printError(
+        'An error occurred when adding LLDB Init File:\n'
+        '  $e',
+      );
+    }
+  }
+
+  Future<SchemeInfo> _getSchemeInfo() async {
+    final XcodeProjectInfo? projectInfo = await _xcodeProject.projectInfo();
+    if (projectInfo == null) {
+      throw Exception('Unable to get Xcode project info.');
+    }
+    if (_xcodeProject.xcodeWorkspace == null) {
+      throw Exception('Xcode workspace not found.');
+    }
+    final String? scheme = projectInfo.schemeFor(_buildInfo);
+    if (scheme == null) {
+      projectInfo.reportFlavorNotFoundAndExit();
+    }
+
+    final File schemeFile = _xcodeProject.xcodeProjectSchemeFile(scheme: scheme);
+    if (!schemeFile.existsSync()) {
+      throw Exception('Unable to get scheme file for $scheme.');
+    }
+
+    final String schemeContent = schemeFile.readAsStringSync();
+    return SchemeInfo(schemeName: scheme, schemeFile: schemeFile, schemeContent: schemeContent);
+  }
+
+  bool _isSchemeMigrated(SchemeInfo schemeInfo) {
+    if (schemeInfo.schemeContent.contains(initPath)) {
+      return true;
+    }
+    return false;
+  }
+
+  void _migrateScheme(SchemeInfo schemeInfo) {
+    final File schemeFile = schemeInfo.schemeFile;
+    final String schemeContent = schemeInfo.schemeContent;
+
+    if (schemeContent.contains('customLLDBInitFile')) {
+      throw Exception(
+        'Running Flutter in debug mode on new iOS versions requires a LLDB '
+        'Init File, but the scheme already has one set. Please remove the LLDB '
+        'Init File for the scheme ${schemeInfo.schemeName}.',
```

Comment:
I wonder if we need to support situations where users have custom LLDB Init File's configured? I doubt this is a very used feature (as it is fairly low-level), but we could also suggest editing LLDB Init File if there is one already, e.g we can suggest adding:

```
command script import .../lldb_helper.py
```

to it or something like that. 

But probably it is uncommon enough so that we can ignore it.